README.md

```diff
@@ -10,6 +10,60 @@ If there is anything you are looking for that is not covered here you should ref
 
 ----
 
+### File structure ###
+
+You should not have multiple types in the same file, each type should have its own file.
```

Comment:
that would be great to define what a type is, just to make sure everybody is on the same page. Eg instead of "have multiple types" maybe say "define multiple types (ie class, struct)"